eos/meson-mixing/bq-mixing.cc

```diff
@@ -0,0 +1,121 @@
+/* vim: set sw=4 sts=4 et foldmethod=syntax : */
+
+/*
+ * Copyright (c) 2021 Danny van Dyk
+ *
+ * This file is part of the EOS project. EOS is free software;
+ * you can redistribute it and/or modify it under the terms of the GNU General
+ * Public License version 2, as published by the Free Software Foundation.
+ *
+ * EOS is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc., 59 Temple
+ * Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+
+#include <eos/meson-mixing/bq-mixing.hh>
+#include <eos/utils/destringify.hh>
+#include <eos/utils/model.hh>
+#include <eos/utils/power_of.hh>
+#include <eos/utils/private_implementation_pattern-impl.hh>
+#include <eos/utils/options-impl.hh>
+
+namespace eos
+{
+    using std::norm;
+
+    template <>
+    struct Implementation<BMixing>
+    {
+        std::shared_ptr<Model> model;
+
+        UsedParameter hbar;
+
+        UsedParameter g_fermi;
+
+        SwitchOption opt_q;
+
+        UsedParameter m_B;
+
+        UsedParameter f_B;
+
+        UsedParameter tau_B;
+
+        UsedParameter R_1;
+        UsedParameter R_2;
+        UsedParameter R_3;
+        UsedParameter R_4;
+        UsedParameter R_5;
+
+        Implementation(const Parameters & p, const Options & o, ParameterUser & u) :
+            model(Model::make(o.get("model", "SM"), p, o)),
+            hbar(p["QM::hbar"], u),
+            g_fermi(p["WET::G_Fermi"], u),
+            opt_q(o, "q", {"s"}),
+            m_B(p["mass::B_" + opt_q.value()], u),
+            f_B(p["decay-constant::B_" + opt_q.value()], u),
+            tau_B(p["life_time::B_" + opt_q.value()], u),
+            R_1(p["B_s<->Bbar_s::R^1"], u),
+            R_2(p["B_s<->Bbar_s::R^2"], u),
+            R_3(p["B_s<->Bbar_s::R^3"], u),
+            R_4(p["B_s<->Bbar_s::R^4"], u),
+            R_5(p["B_s<->Bbar_s::R^5"], u)
+        {
+            u.uses(*model);
+        }
+
+        complex<double> M_12() const
+        {
+            const double mu = 4.2;
```

Comment:
Hard-coded scale - maybe allow to set this via some option to some other value? Of course this would also require to do the RG evolution the R_1,...,5 to that new value of mu. At least this is something to keep in mind for the future. Currently the input for R_1,....,5 should be provided at mu = 4.2 GeV.

Reply:
I agree that this might needs changing at some point, but I would deem it good enough for now.